--- a/src/arch/x86_64/CodeGen.zig
+++ b/src/arch/x86_64/CodeGen.zig
@@ -83608,7 +83608,7 @@
 
                     const lhs_mcv = try self.resolveInst(bin_op.lhs);
                     const mat_lhs_mcv = switch (lhs_mcv) {
-                        .load_symbol => mat_lhs_mcv: {
+                        .load_symbol, .load_direct, .lea_direct => mat_lhs_mcv: {
                             // TODO clean this up!
                             const addr_reg = try self.copyToTmpRegister(.usize, lhs_mcv.address());
                             break :mat_lhs_mcv MCValue{ .indirect = .{ .reg = addr_reg } };
@@ -83632,7 +83632,7 @@
 
                     const rhs_mcv = try self.resolveInst(bin_op.rhs);
                     const mat_rhs_mcv = switch (rhs_mcv) {
-                        .load_symbol => mat_rhs_mcv: {
+                        .load_symbol, .load_direct, .lea_direct => mat_rhs_mcv: {
                             // TODO clean this up!
                             const addr_reg = try self.copyToTmpRegister(.usize, rhs_mcv.address());
                             break :mat_rhs_mcv MCValue{ .indirect = .{ .reg = addr_reg } };
@@ -83733,7 +83733,7 @@
 
                         const rhs_mcv = try self.resolveInst(bin_op.rhs);
                         const mat_rhs_mcv = switch (rhs_mcv) {
-                            .load_symbol => mat_rhs_mcv: {
+                            .load_symbol, .load_direct, .lea_direct => mat_rhs_mcv: {
                                 // TODO clean this up!
                                 const addr_reg = try self.copyToTmpRegister(.usize, rhs_mcv.address());
                                 break :mat_rhs_mcv MCValue{ .indirect = .{ .reg = addr_reg } };
@@ -83965,7 +83965,7 @@
 
             const lhs_mcv = try self.resolveInst(bin_op.lhs);
             const mat_lhs_mcv = switch (lhs_mcv) {
-                .load_symbol => mat_lhs_mcv: {
+                .load_symbol, .load_direct, .lea_direct => mat_lhs_mcv: {
                     // TODO clean this up!
                     const addr_reg = try self.copyToTmpRegister(.usize, lhs_mcv.address());
                     break :mat_lhs_mcv MCValue{ .indirect = .{ .reg = addr_reg } };
@@ -83989,7 +83989,7 @@
 
             const rhs_mcv = try self.resolveInst(bin_op.rhs);
             const mat_rhs_mcv = switch (rhs_mcv) {
-                .load_symbol => mat_rhs_mcv: {
+                .load_symbol, .load_direct, .lea_direct => mat_rhs_mcv: {
                     // TODO clean this up!
                     const addr_reg = try self.copyToTmpRegister(.usize, rhs_mcv.address());
                     break :mat_rhs_mcv MCValue{ .indirect = .{ .reg = addr_reg } };
@@ -84520,7 +84520,7 @@
                                 break :mat_lhs_mcv mat_lhs_mcv;
                             },
                         },
-                        .load_symbol => {
+                        .load_symbol, .load_direct, .lea_direct => {
                             // TODO clean this up!
                             const addr_reg = try self.copyToTmpRegister(.usize, lhs_mcv.address());
                             break :mat_lhs_mcv MCValue{ .indirect = .{ .reg = addr_reg } };
@@ -84544,7 +84544,7 @@
                                 break :mat_rhs_mcv mat_rhs_mcv;
                             },
                         },
-                        .load_symbol => {
+                        .load_symbol, .load_direct, .lea_direct => {
                             // TODO clean this up!
                             const addr_reg = try self.copyToTmpRegister(.usize, rhs_mcv.address());
                             break :mat_rhs_mcv MCValue{ .indirect = .{ .reg = addr_reg } };
@@ -89323,7 +89323,7 @@
         defer for (reg_locks) |reg_lock| if (reg_lock) |lock| self.register_manager.unlockReg(lock);
 
         const mat_lhs_mcv = switch (lhs_mcv) {
-            .load_symbol => mat_lhs_mcv: {
+            .load_symbol, .load_direct, .lea_direct => mat_lhs_mcv: {
                 // TODO clean this up!
                 const addr_reg = try self.copyToTmpRegister(.usize, lhs_mcv.address());
                 break :mat_lhs_mcv MCValue{ .indirect = .{ .reg = addr_reg } };
@@ -89336,7 +89336,7 @@
         };
         defer if (mat_lhs_lock) |lock| self.register_manager.unlockReg(lock);
         const mat_rhs_mcv = switch (rhs_mcv) {
-            .load_symbol => mat_rhs_mcv: {
+            .load_symbol, .load_direct, .lea_direct => mat_rhs_mcv: {
                 // TODO clean this up!
                 const addr_reg = try self.copyToTmpRegister(.usize, rhs_mcv.address());
                 break :mat_rhs_mcv MCValue{ .indirect = .{ .reg = addr_reg } };
