diff --git a/Tests/LibIPC/TestConnection.cpp b/Tests/LibIPC/TestConnection.cpp
index 8f05adaf32..865dfb9376 100644
--- a/Tests/LibIPC/TestConnection.cpp
+++ b/Tests/LibIPC/TestConnection.cpp
@@ -13,7 +13,7 @@
 #include <LibIPC/Connection.h>
 #include <LibIPC/Message.h>
 #include <LibIPC/Stub.h>
-#include <LibIPC/TransportSocket.h>
+#include <LibIPC/Transport.h>
 #include <LibTest/TestCase.h>
 
 namespace {
@@ -106,7 +106,7 @@ TEST_CASE(sync_wait_failure_does_not_dispatch_queued_messages)
     MUST(local_socket->set_blocking(false));
     MUST(peer_socket->set_blocking(false));
 
-    auto transport = make<IPC::TransportSocket>(move(local_socket));
+    auto transport = make<IPC::Transport>(move(local_socket));
 
     CountingStub stub;
     auto connection = TestConnection::construct(stub, move(transport));
diff --git a/Utilities/js.cpp b/Utilities/js.cpp
index 716225f959..da2fe2e1e6 100644
--- a/Utilities/js.cpp
+++ b/Utilities/js.cpp
@@ -36,7 +36,7 @@
 #include <LibMain/Main.h>
 #include <LibTextCodec/Decoder.h>
 
-#if !defined(AK_OS_WINDOWS) && !defined(AK_OS_ANDROID)
+#if !defined(AK_OS_WINDOWS) && !defined(AK_OS_ANDROID) && !defined(AK_OS_PLAN9)
 #    include <editline/readline.h>
 #endif
 #include <stdlib.h>
@@ -101,7 +101,7 @@ static bool s_print_last_result = false;
 static bool s_strip_ansi = false;
 static bool s_raw_strings = false;
 static bool s_disable_source_location_hints = false;
-#if !defined(AK_OS_WINDOWS) && !defined(AK_OS_ANDROID)
+#if !defined(AK_OS_WINDOWS) && !defined(AK_OS_ANDROID) && !defined(AK_OS_PLAN9)
 static JS::Realm* s_repl_realm;
 static JS::GlobalEnvironment* s_repl_global_environment;
 #endif
@@ -512,7 +512,7 @@ private:
     int m_group_stack_depth { 0 };
 };
 
-#if !defined(AK_OS_WINDOWS) && !defined(AK_OS_ANDROID)
+#if !defined(AK_OS_WINDOWS) && !defined(AK_OS_ANDROID) && !defined(AK_OS_PLAN9)
 static Vector<ByteString> complete_repl_line(StringView line)
 {
     auto& realm = *s_repl_realm;
@@ -891,7 +891,7 @@ ErrorOr<int> ladybird_main(Main::Arguments arguments)
     // FIXME: Figure out some way to interrupt the interpreter now that vm.exception() is gone.
 
     if (evaluate_script.is_empty() && script_paths.is_empty()) {
-#if defined(AK_OS_WINDOWS) || defined(AK_OS_ANDROID)
+#if defined(AK_OS_WINDOWS) || defined(AK_OS_ANDROID) || defined(AK_OS_PLAN9)
         dbgln("REPL functionality is not supported on this platform");
         VERIFY_NOT_REACHED();
 #else
