diff --git a/Libraries/LibCore/System.h b/Libraries/LibCore/System.h
index 3b2de6a921..e4576c227e 100644
--- a/Libraries/LibCore/System.h
+++ b/Libraries/LibCore/System.h
@@ -46,7 +46,7 @@ struct addrinfo;
 struct sockaddr;
 #endif
 
-#if !defined(AK_OS_BSD_GENERIC) && !defined(AK_OS_ANDROID) && !defined(AK_OS_WINDOWS)
+#if !defined(AK_OS_BSD_GENERIC) && !defined(AK_OS_ANDROID) && !defined(AK_OS_WINDOWS) && !defined(AK_OS_PLAN9)
 #    include <shadow.h>
 #endif
 
